probe-rs/src/bin/probe-rs/cmd/run.rs

```diff
@@ -112,18 +112,21 @@ impl Cmd {
                 }
             }
         }
-        core.run()?;
-
-        run_loop(
-            &mut core,
-            &memory_map,
-            &rtt_scan_regions,
-            path,
-            timestamp_offset,
-            self.always_print_stacktrace,
-            self.no_location,
-            self.log_format.as_deref(),
-        )?;
+
+        if core.core_halted()? {
+            core.run()?;
+        } else {
```

Comment:
Why the `else`? IIUC right now if the core is halted, probe-rs starts the core and just exits instead of polling for logs and panics.

Reply:
thanks for pointing it out! you are right, it shouldn't be inside a "else", which kinda breaks "probe-rs run" cmd.